### ISSUE TITLE
Default header documentation is incomplete

### ISSUE THREAD

**Issue by gilly3** [axios/axios#2436]
**Section/Content To Improve**
[Config defaults](https://github.com/axios/axios#config-defaults)

**Suggested Improvement**
The documentation teases with a concept of `common` when setting default header values:

```js
axios.defaults.headers.common['Authorization'] = AUTH_TOKEN;
axios.defaults.headers.post['Content-Type'] = 'application/x-www-form-urlencoded';
```

Presumably, that means for _all_ request methods, but (playing devil's advocate here) maybe it's demonstrating that you can set default headers for custom request methods (like if your server accepts a custom request method of `common`).

Is `common` actually a keyword that means any request method?

Are there other keywords with special meaning?  Like, for example, one that applies only to requests containing a payload?

Does any other non-keyword value apply to a `method` value?  Even custom methods?

Is it possible to set a default header for requests to my server that understands `common` as a custom HTTP method?

If I set a `common` header and a `get` header, does `get` override `common`?

Is there a simple way to apply a header to requests of every method type except one?

```js
axios.defaults.headers.common['Content-Type'] = 'application/json';
axios.defaults.headers.get['Content-Type'] = null;
```

What effect will that have?

Some of my questions are hypothetical.  Forgive me.

**Relevant File(s)**:  README.md

**Comment by yasuf** [comment]
> Is common actually a keyword that means any request method?

`common` is a word used in this library used for the headers that will be set by default on a request with any verb, you can see how this works here, when you have different headers for `common` and `get` for example, the `get` headers take priority, otherwise, the `header` property is used as the object to be used which contains all the headers.

https://github.com/axios/axios/blob/405d50ecf494864b5dc428775aaa2ec347393726/lib/core/dispatchRequest.js#L37-L42

> Is there a simple way to apply a header to requests of every method type except one?

You can set the headers for `common` and set it to null or an empty string in the verb that you're looking to exclude.

Let me know if this is helpful

**Comment by chinesedfan** [comment]
Closed in favor of #2618.
